### PR TITLE
Create Github Issue and Pull Request templates

### PR DESCRIPTION
**Pull request by mikejbrown**
These templates are automatically filled in the editor textarea when
someone creates an Issue or Pull request on Github.
The text can be changed or deleted at will by the user, it is there merely
to serve as a prompt to include the pertinent information and format it in
a useful way.

Here is an example of how this PR might look if using the template:

----

### The issue this fixes

Issues and pull request messages have been of very uneven quality. I am as guilty as anyone in terms of including inadequate or poorly presented information in pull requests. But over the history of the project so far, people have submitted things ranging from impossible to read stream of consciousness style messages, to messages with no informative content at all.

### What this PR does

Luckily Github makes it possible to create [issue](https://help.github.com/articles/creating-an-issue-template-for-your-repository/) and [pull request](https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/) templates to make it easier for contributors to write meaningful messages that are informative and follow the contributing guidelines for the project.

I've taken the liberty of creating some templates based on the contributing guidelines and discord feedback.

Pings: Thanks @frankitox16 @bjubes and @TomMalbran for useful feedback via discord.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+Fixes # 
+Closes # 
+
+### What this PR does
+
+Describe _how_ you have fixed the issue(s).
+If there are any UI or sprite changes, please include an image showing the new element(s).
+Please ensure that you have the proper permission to include any copyrighted art or sound assets under license terms compatible with the open source licensing of this project.
+
+## TODO (if this is a [WIP] PR)
```
Comment (bjubes) [inline .github/PULL_REQUEST_TEMPLATE.md]: Todo is the wrong header size compared the to the rest of the template

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Briefly describe the issue(s) addressed by this pull request. If there are any relevant issues or pull requests on Github reference their #numbers here.
+
+Fixes # 
+Closes # 
+
+### What this PR does
+
+Describe _how_ you have fixed the issue(s).
+If there are any UI or sprite changes, please include an image showing the new element(s).
```
Comment (TomMalbran) [inline .github/PULL_REQUEST_TEMPLATE.md]: Could we add a note saying something like that "After doing significant changes to the original design and when you are ready for a new review of the PR, add a new image with the new look"?

For animations you can include a gif. Should this be required?

Comment (bjubes) [inline .github/PULL_REQUEST_TEMPLATE.md]: "Include an image and update it for subsequent changes"

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (mikejbrown) [inline .github/PULL_REQUEST_TEMPLATE.md]: Good catch

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (mikejbrown) [inline .github/PULL_REQUEST_TEMPLATE.md]: How does it look now?

Comment (TomMalbran) [inline .github/PULL_REQUEST_TEMPLATE.md]: Looks good,

**Comment by TomMalbran** [comment]
One final question. Should we add a comment in the issues saying that you can add images to better visualize the issue, but not required?

**Comment by mikejbrown** [comment]
@TomMalbran Like that? @bjubes Still approve?

**Comment by koosemose** [comment]
This seems to have reached a consensus state, Merging.

**Pull request merged**